From 7ca572992a8a16b7fccb0f04be0c38c3d34019cc Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Fri, 12 Sep 2014 18:10:12 -0400
Subject: [powerpc] Adjust CPU_FTR_SMT on all platforms

Message-id: <1410545655-205645-384-git-send-email-dzickus@redhat.com>
Patchwork-id: 94357
O-Subject: [RHEL7 PATCH 383/626] powerpc: Adjust CPU_FTR_SMT on all platforms
Bugzilla: 1127366
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: David Gibson <dgibson@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1127366

commit 4a85b31d369b496c316d89b144ee9626073e5ef2
Author: Benjamin Herrenschmidt <benh@kernel.crashing.org>
Date:   Fri Mar 28 13:36:26 2014 +1100

    powerpc: Adjust CPU_FTR_SMT on all platforms

    For historical reasons that code was under #ifdef CONFIG_PPC_PSERIES
    but it applies equally to all 64-bit platforms.

    Signed-off-by: Benjamin Herrenschmidt <benh@kernel.crashing.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/kernel/prom.c b/arch/powerpc/kernel/prom.c
index ae5b890..00fdad0 100644
--- a/arch/powerpc/kernel/prom.c
+++ b/arch/powerpc/kernel/prom.c
@@ -378,7 +378,7 @@ static int __init early_init_dt_scan_cpus(unsigned long node,
  check_cpu_pa_features(node);
  check_cpu_slb_size(node);
 
-#ifdef CONFIG_PPC_PSERIES
+#ifdef CONFIG_PPC64
  if (nthreads > 1)
   cur_cpu_spec->cpu_features |= CPU_FTR_SMT;
  else
-- 
1.7.1